Makefile

```diff
@@ -2,30 +2,42 @@ PREFIX            ?= $(shell pwd)
 FILES             ?= $(shell find . -type f -name '*.go' -not -path "./vendor/*")
 DOCKER_IMAGE_NAME ?= thanos
 DOCKER_IMAGE_TAG  ?= $(subst /,-,$(shell git rev-parse --abbrev-ref HEAD))-$(shell date +%Y-%m-%d)-$(shell git rev-parse --short HEAD)
+FIRST_GOPATH      ?= $(firstword $(subst :, ,$(shell go env GOPATH)))
```

Comment:
Can we have some comment why First GOPATH and not the last one or just leave the system to choose proper one?

Reply:
`$GOPATH/bin` might not be in `$PATH`, so I couldn't assume `which` would give me the path of promu, dep et al. As for selecting the first GOPATH, I admit it's a bit of an assumption:
- most people only have one GOPATH at a time;
- if you don't have one or any of those tools installed, running `go get` would place them in the first GOPATH.
It's possible that any of the tools would be installed in the other GOPATHs, but for simplicity sake we just make sure they exist in the first one, and them keep using those.

Reply:
Wow, this seems like lots of assumptions are there. 
> $GOPATH/bin might not be in $PATH

Right, but it should. https://golang.org/doc/code.html#GOPATH (for "convention")

Anyway, we can start with something like this, but please place a comment before FIRST_GOPATH with what you mentioned here. (:

Reply:
Sure.